### DISCUSSION TITLE
What is the maximum number of vertices in a mesh?

### DISCUSSION THREAD

**Discussion by JonahPlusPlus** [bevyengine/bevy#2950] [Q&A] [answered]
I'm thinking of making a prototype battle royale and was looking at Bevy, since I plan on making it P2P, and it seems easier to write the network code in Rust than in C# in Unity (seems like it would get messy). That being said, it would force me to create the terrain in Blender. 

So, I'm wondering what the maximum number of vertices is in a mesh, in case I can get away with making one huge model (a huge mountain, so not too sprawled out), or if I will have to split the mesh and then load each part in (better for performance). 

And could I make it into one GLTF/GLB file and be able to put the objects into an octree and cull objects somehow? Or does Bevy do culling under the hood, so I don't have to?

**Comment by tofoz** [comment, marked as answer]
I believe you can use u16 or u32 indices, so a max number of vertices would be 65,535 or 4,294,967,295.

**Reply by JonahPlusPlus** [reply]
That makes sense, thanks!
